libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -74,6 +74,8 @@ namespace eosio { namespace chain {
             validation_mode          block_validation_mode  = validation_mode::FULL;
 
             flat_set<account_name>   resource_greylist;
+            flat_set<account_name>   trusted_producers;
+            bool                     trusted_producer_light_validation = false;
```

Comment:
I dont really like this "implementation detail" exposed in the `config` structure (The `bool`,  the `flat_set` is appropriate here).  There doesn't seem to be a reason why an external system would ever need to know about this.  I think I would prefer that be part of the `controller_impl` class instead alongside `in_trx_requiring_checks`

Reply:
also `trusted_producers` should be added to the reflection of this structure: https://github.com/EOSIO/eos/blob/c7a5a9e9e8c775cb125571db484aef8db9b71aba/libraries/chain/include/eosio/chain/controller.hpp#L294